drivers/interrupt_controller/intc_plic.c

```diff
@@ -48,18 +49,44 @@ static inline uint32_t get_plic_enabled_size(const struct device *dev)
 	return (config->num_irqs >> 5) + 1;
 }
 
+/* Forward declaration */
+#define PLIC_INTC_DEV_CONFIG_DECL(n) static const struct plic_config plic_config_##n
+DT_INST_FOREACH_STATUS_OKAY(PLIC_INTC_DEV_CONFIG_DECL);
+
+#define PLIC_GET_DEV_FROM_IRQ(n)                                                                   \
+	do {                                                                                       \
+		if (irq < (plic_config_##n.isr_offset + plic_config_##n.num_irqs)) {               \
```

Comment:
probably safer to check that irq is between `plic_config_##n.isr_offset` & `plic_config_##n.isr_offset + plic_config_##n.num_irqs`